ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -727,17 +727,19 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
     }
 
     addressIsBlank(address: Address): boolean {
-        return !address || this.countOfFieldsWithData(address) === 0;
+        return !address || !this.countOfFieldsWithData(address);
     }
 
     enoughDataToSave(address: Address | null): boolean {
-        return address && this.countOfFieldsWithData(address) >= 1;
+        return address && this.countOfFieldsWithData(address);
     }
 
-    countOfFieldsWithData(address: Address): number {
-        const isNonBlankString: FuncType<boolean> = (val) => util.isString(val) && val.trim().length > 0;
-        const propsToCheck: (keyof Address)[] = ['name', 'country', 'street1', 'street2', 'state', 'city', 'zip', 'phone'];
-        return propsToCheck.map(prop => address[prop]).filter((value) => isNonBlankString(value)).length;
+    countOfFieldsWithData(address: Address): boolean {
+        return Object.entries(address).every(([key, value]) =>
+            !['street2', 'phone', 'guid'].includes(key)
```

Comment:
Previously we checked a defined list of address fields and whether all of them are not empty.

1. Why do you exclude 'street2` and `phone` form that ^ check? It is not obvious for me.

Reply:
2. The check is supposed to go through all the address properties (except `'street2', 'phone', 'guid'`).
There is also boolean field 'isDefault' there. 
It passes the check (`value.toString().trim().length > 0`) in any case 
But why should we check it at all?

Reply:
Street2 and phone is not required fields that's why we are excluding it, but adding logic so that if phone is required it's not excluded, will update pr soon.
about isDefault property - that becomes string 'true' and so passes the test but sure we can exlude it as well.